embedded/src/main/java/org/wildfly/core/embedded/EmbeddedProcessFactory.java

```diff
@@ -353,30 +353,37 @@ private static Object createManagedProcess(final ProcessType embeddedType, final
 
     private static void resetEmbeddedServerProperties(final String jbossHomeDir, final ProcessType embeddedServerType) {
         assert jbossHomeDir != null;
-
+        String oldJbossHomeDir = WildFlySecurityManager.getPropertyPrivileged(SYSPROP_KEY_JBOSS_HOME_DIR, null);
         String jbossBaseDir;
+        boolean shouldReset = oldJbossHomeDir != null && !jbossHomeDir.equalsIgnoreCase(oldJbossHomeDir);
```

Comment:
I'm not sure we should be ignoring the case here. Path names can be case sensitive.